### PR TITLE
Fix 'Join to Kubernetes cluster' task

### PR DESCRIPTION
**Pull request by atsikham**
Copying is handled by Ansible's 'copy' module. From [documentation](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/copy_module.html#parameter-force):

```text
If yes, the remote file will be replaced when contents are different than the source.
```

So no checksums are checked, only condition removed.

### CODE REVIEW

**Review by seriva — DISMISSED** [review]

**Review by cicharka — DISMISSED** [review]

**Review by seriva — DISMISSED** [review]

**Review by to-bar — DISMISSED** [review]

**Comment by przemyslavic** [comment]
@atsikham 
It doesn't solve the issue described in the task #1175.
1. Deployed k8s cluster (1 master + 1 node)
2. Executed `kubeadm reset` on master node
3. Ran `epicli apply` again and it failed:
```
2021-12-29T11:27:03.6324390Z[38;21m11:27:03 INFO cli.engine.ansible.AnsibleCommand - TASK [kubernetes_node : Join to Kubernetes cluster] ****************************
2021-12-29T11:27:04.6504789Z[31;21m11:27:04 ERROR cli.engine.ansible.AnsibleCommand - fatal: [ci-pkiazurcentcanal-kubernetes-node-vm-0]: FAILED! => {"changed": true, "cmd": "kubeadm join  --config /etc/kubeadm/kubeadm-join-node.yml\n", "delta": "0:00:00.392559", "end": "2021-12-29 11:27:04.577311", "msg": "non-zero return code", "rc": 1, "start": "2021-12-29 11:27:04.184752", "stderr": "error execution phase preflight: [preflight] Some fatal errors occurred:\n\t[ERROR FileAvailable--etc-kubernetes-kubelet.conf]: /etc/kubernetes/kubelet.conf already exists\n\t[ERROR Port-10250]: Port 10250 is in use\n\t[ERROR FileAvailable--etc-kubernetes-pki-ca.crt]: /etc/kubernetes/pki/ca.crt already exists\n[preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`\nTo see the stack trace of this error execute with --v=5 or higher", "stderr_lines": ["error execution phase preflight: [preflight] Some fatal errors occurred:", "\t[ERROR FileAvailable--etc-kubernetes-kubelet.conf]: /etc/kubernetes/kubelet.conf already exists", "\t[ERROR Port-10250]: Port 10250 is in use", "\t[ERROR FileAvailable--etc-kubernetes-pki-ca.crt]: /etc/kubernetes/pki/ca.crt already exists", "[preflight] If you know what you are doing, you can make a check non-fatal with `--ignore-preflight-errors=...`", "To see the stack trace of this error execute with --v=5 or higher"], "stdout": "[preflight] Running pre-flight checks", "stdout_lines": ["[preflight] Running pre-flight checks"]}
2021-12-29T11:27:04.6649538Z[38;21m11:27:04 INFO cli.engine.ansible.AnsibleCommand - 
2021-12-29T11:27:04.6650931Z[38;21m11:27:04 INFO cli.engine.ansible.AnsibleCommand - TASK [kubernetes_node : Join to cluster with ignores] **************************
2021-12-29T11:29:05.9094641Z[31;21m11:29:05 ERROR cli.engine.ansible.AnsibleCommand - fatal: [ci-pkiazurcentcanal-kubernetes-node-vm-0]: FAILED! => {"changed": true, "cmd": "kubeadm join  --config /etc/kubeadm/kubeadm-join-node.yml  --ignore-preflight-errors all\n", "delta": "0:02:00.670749", "end": "2021-12-29 11:29:05.816895", "msg": "non-zero return code", "rc": 1, "start": "2021-12-29 11:27:05.146146", "stderr": "\t[WARNING FileAvailable--etc-kubernetes-kubelet.conf]: /etc/kubernetes/kubelet.conf already exists\n\t[WARNING Port-10250]: Port 10250 is in use\n\t[WARNING FileAvailable--etc-kubernetes-pki-ca.crt]: /etc/kubernetes/pki/ca.crt already exists\nerror execution phase kubelet-start: error uploading crisocket: timed out waiting for the condition\nTo see the stack trace of this error execute with --v=5 or higher", "stderr_lines": ["\t[WARNING FileAvailable--etc-kubernetes-kubelet.conf]: /etc/kubernetes/kubelet.conf already exists", "\t[WARNING Port-10250]: Port 10250 is in use", "\t[WARNING FileAvailable--etc-kubernetes-pki-ca.crt]: /etc/kubernetes/pki/ca.crt already exists", "error execution phase kubelet-start: error uploading crisocket: timed out waiting for the condition", "To see the stack trace of this error execute with --v=5 or higher"], "stdout": "[preflight] Running pre-flight checks\n[preflight] Reading configuration from the cluster...\n[preflight] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -o yaml'\n[kubelet-start] Writing kubelet configuration to file \"/var/lib/kubelet/config.yaml\"\n[kubelet-start] Writing kubelet environment file with flags to file \"/var/lib/kubelet/kubeadm-flags.env\"\n[kubelet-start] Starting the kubelet\n[kubelet-start] Waiting for the kubelet to perform the TLS Bootstrap...\n[kubelet-check] Initial timeout of 40s passed.", "stdout_lines": ["[preflight] Running pre-flight checks", "[preflight] Reading configuration from the cluster...", "[preflight] FYI: You can look at this config file with 'kubectl -n kube-system get cm kubeadm-config -o yaml'", "[kubelet-start] Writing kubelet configuration to file \"/var/lib/kubelet/config.yaml\"", "[kubelet-start] Writing kubelet environment file with flags to file \"/var/lib/kubelet/kubeadm-flags.env\"", "[kubelet-start] Starting the kubelet", "[kubelet-start] Waiting for the kubelet to perform the TLS Bootstrap...", "[kubelet-check] Initial timeout of 40s passed."]}
2021-12-29T11:29:05.9296216Z[38;21m11:29:05 INFO cli.engine.ansible.AnsibleCommand - 
2021-12-29T11:29:05.9297310Z[38;21m11:29:05 INFO cli.engine.ansible.AnsibleCommand - TASK [kubernetes_node : Display kubeadm join stderr if any] ********************
2021-12-29T11:29:05.9885416Z[38;21m11:29:05 INFO cli.engine.ansible.AnsibleCommand - ok: [ci-pkiazurcentcanal-kubernetes-node-vm-0] => {
2021-12-29T11:29:05.9887711Z[31;21m11:29:05 ERROR cli.engine.ansible.AnsibleCommand -     "msg": "Joined with warnings\n['\\t[WARNING FileAvailable--etc-kubernetes-kubelet.conf]: /etc/kubernetes/kubelet.conf already exists', '\\t[WARNING Port-10250]: Port 10250 is in use', '\\t[WARNING FileAvailable--etc-kubernetes-pki-ca.crt]: /etc/kubernetes/pki/ca.crt already exists', 'error execution phase kubelet-start: error uploading crisocket: timed out waiting for the condition', 'To see the stack trace of this error execute with --v=5 or higher']\n"
2021-12-29T11:29:05.9897007Z[38;21m11:29:05 INFO cli.engine.ansible.AnsibleCommand - }
2021-12-29T11:29:06.0065788Z[38;21m11:29:06 INFO cli.engine.ansible.AnsibleCommand - 
2021-12-29T11:29:06.0066806Z[38;21m11:29:06 INFO cli.engine.ansible.AnsibleCommand - TASK [kubernetes_node : Mark node regardless of join result] *******************
2021-12-29T11:29:06.0535320Z[38;21m11:29:06 INFO cli.engine.ansible.AnsibleCommand - ok: [ci-pkiazurcentcanal-kubernetes-node-vm-0]
2021-12-29T11:29:06.0564255Z[38;21m11:29:06 INFO cli.engine.ansible.AnsibleCommand - 
2021-12-29T11:29:06.0572498Z[38;21m11:29:06 INFO cli.engine.ansible.AnsibleCommand - PLAY RECAP *********************************************************************
2021-12-29T11:29:06.0579407Z[38;21m11:29:06 INFO cli.engine.ansible.AnsibleCommand - ci-pkiazurcentcanal-kubernetes-master-vm-0 : ok=9    changed=0    unreachable=0    failed=0    skipped=2    rescued=0    ignored=0   
2021-12-29T11:29:06.0584014Z[38;21m11:29:06 INFO cli.engine.ansible.AnsibleCommand - ci-pkiazurcentcanal-kubernetes-node-vm-0 : ok=37   changed=1    unreachable=0    failed=1    skipped=12   rescued=1    ignored=0   
2021-12-29T11:29:06.0587266Z[38;21m11:29:06 INFO cli.engine.ansible.AnsibleCommand - ci-pkiazurcentcanal-repository-vm-0 : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

**Comment by przemyslavic** [comment]
:heavy_check_mark: Executed `kubeadm reset` on each node and it succeeded.
The question is, do we want the reset of each node to be somehow automated when something broke, or users should be responsible for it before running `apply` again.

Also task #2669 seems to be related.

File: ansible/playbooks/roles/kubernetes_node/tasks/node-join.yml [hunk 1]
```diff
@@ -30,23 +30,29 @@
...
+            - name: Attempt to join with kubeadm reset
+              when: kubeadm_join_result.stderr is search('/etc/kubernetes/kubelet.conf already exists')
+              block:
+                - name: Reset node
+                  command: kubeadm reset --force
+
+                - <<: *soft-join
+
+              rescue:
+                - name: Join to cluster with ignores
```
Comment (to-bar) [inline ansible/playbooks/roles/kubernetes_node/tasks/node-join.yml:50]: Please create an issue for removing (or adding comment about) `--ignore-preflight-errors all` and add the issue number here as a comment.

**Review by to-bar — DISMISSED** [review]

**Review by seriva — APPROVED** [review]

**Review by przemyslavic — APPROVED** [review]

**Comment by przemyslavic** [comment]
Tested scenario:
1. Deployed k8s cluster: 1 master + 1 node
2. Ran kubeadm reset on master node
3. Executed epicli apply again and it succeedeed
 Tested on AWS Ubuntu/RHEL/CentOS

File: ansible/playbooks/roles/kubernetes_node/tasks/node-join.yml (hunk 1 above)
Comment (atsikham) [reply ansible/playbooks/roles/kubernetes_node/tasks/node-join.yml:50]: #2828

**Comment by przemyslavic** [comment]
Commented on a PR https://github.com/epiphany-platform/epiphany/pull/2805.

**Pull request merged**